### DISCUSSION TITLE
How to upgrade Solarium library on Drupal 10

### DISCUSSION THREAD

**Discussion by BrzIGH** [solariumphp/solarium#1123] [Q&A]
I have updated the Solr Search API and my Drupal status page is tellign me that Solarium 6.3.3 is out of date and I need Solarium 6.3.5.
What is the best way to perform this upgrade ?
Note that my Solr search works fine as it is...
Thanks

**Comment by thomascorthals** [comment]
Looks like Drupal's [Composer Manager](https://www.drupal.org/project/composer_manager) is what you need. (Disclaimer: never worked with it myself.)

You won't notice any changes in the Solr search between Solarium 6.3.3 and 6.3.5 unless you use facet pivots or suggesters. Check the [CHANGELOG](https://github.com/solariumphp/solarium/blob/master/CHANGELOG.md) if you think this might affect you.

**Comment by BrzIGH** [comment]
Thanks

Composer manager is for Drupal 6,7 and 8. I am running Drupal 10. Not sure I want to risk that.
Will a straight manual library file overwrite work ?

**Comment by thomascorthals** [comment]
In general, this should be safe with Composer.

```sh
cp composer.lock composer.lock~
composer update solarium/solarium
```

If it does go wrong, you can revert to the old situation.

```sh
mv composer.lock~ composer.lock
composer install
```

**Reply by BrzIGH** [reply]
My composer.lock has this section…

    "require": {
                "composer-runtime-api": ">=2.0",
                "composer/semver": "^1.0|^3.0",
                "consolidation/annotated-command": "^2.12|^4.1",
                "drupal/core": "^9.3 || ^10.0",
                "drupal/search_api": "~1.29",
                "ext-dom": "*",
                "ext-json": "*",
                "ext-simplexml": "*",
                "laminas/laminas-stdlib": "^3.2",
                "maennchen/zipstream-php": "^2.2.1|^3.0.2",
                **"solarium/solarium": "^6.3.0"**
                ...etc...

So it looks to me like it will revert to 6.3.0 if I run composer at a later time to update another module….

Am I getting this right.?

Thanks

**Reply by thomascorthals** [reply]
That is your `composer.json`. It lists the requirements for your project. `^6.3.0` means 6.3.0 is the minimum version required and with the `^` it allows updating to any version `>=6.3.0 <7.0.0`.

In `composer.lock` the exact versions of all your installed dependencies are recorded. When you run `composer update`, it updates those versions to the latest versions that are allowed in your `composer.json` and writes a new `composer.lock` file. If you run `composer install` instead, it doesn't update anything and just installs the exact versions it reads from `composer.lock`. That's why you can back it up before an `update` and put it back and `install` from it again to revert to the previous configuration if you run into an issue with an update.

**Reply by BrzIGH** [reply]
Thank you

**Reply by BrzIGH** [reply]
Looks like it worked OK. My searches work fine. My derivatives are created just fine.

I have this text in the Drupal 10 status report :

_Solr Server Solr Server
Schema modified
Your config-set contains manually added customizations. Be aware that these will be lost when the config-set needs to be regenerated._

Can I do this ?

1. Get config.zip from ../admin/config/search/search-api/server/default_solr_server
2. Extract contents and overwrite into /var/solr/data/ISLANDORA/conf
3. Clear all indexed data from ../admin/config/search/search-api/index/default_solr_index (red button)
4. Confirm no indexed documents.
5. Run CRON from Drupal interface (to re-index content)
6. Confirm correct number of documents.

Thanks
